docs/components/nav_drawer.md

```diff
@@ -0,0 +1,324 @@
+              selected: _selectedDestination == 1,
+              onTap: () => selectDestination(1),
+            ),
+            _NavDrawerDestination(
+              iconData: Icons.label,
+              title: 'Item 3',
+              selected: _selectedDestination == 2,
+              onTap: () => selectDestination(2),
+            ),
+            Divider(
+              height: 1,
+              thickness: 1,
+            ),
+            Padding(
+              padding: const EdgeInsets.all(16.0),
+              child: Text(
+                'Label',
+              ),
+            ),
+            _NavDrawerDestination(
+              iconData: Icons.bookmark,
+              title: 'Item A',
+              selected: _selectedDestination == 3,
+              onTap: () => selectDestination(3),
+            ),
+          ],
+        ),
+      ),
+      body: GridView.count(
+        crossAxisCount: 2,
+        crossAxisSpacing: 20,
+        mainAxisSpacing: 20,
+        padding: EdgeInsets.all(20),
+        childAspectRatio: 3 / 2,
+        children: [
+          Image.asset('assets/nav-drawer-1.jpg'),
+          Image.asset('assets/nav-drawer-2.jpg'),
+          Image.asset('assets/nav-drawer-3.jpg'),
+          Image.asset('assets/nav-drawer-4.jpg'),
+        ],
+      ),
+    );
+  }
+
+  void selectDestination(int index) {
+    setState(() {
+      _selectedDestination = index;
+    });
+  }
+}
+
+class _NavDrawerDestination extends StatelessWidget {
+  _NavDrawerDestination({
+    this.iconData,
+    this.title,
+    this.selected,
+    this.onTap,
+  });
+
+  final IconData iconData;
+  final String title;
+  final bool selected;
+  final VoidCallback onTap;
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+    final textTheme = theme.textTheme;
+    final primaryColor = theme.primaryColor;
+
+    return ListTile(
+      leading: Icon(
+        iconData,
+        color: selected ? primaryColor : null,
+      ),
+      title: Text(
+        title,
+        style: textTheme.subtitle2.copyWith(
+          color: selected ? primaryColor : null,
+        ),
+      ),
+      selectedTileColor: primaryColor.withOpacity(0.12),
+      selected: selected,
+      onTap: onTap,
+    );
+  }
+}
+```
+
+## Bottom navigation drawer
+
+Bottom navigation drawers are modal drawers that are anchored to the bottom of the screen instead of the left or right edge. They are only used with bottom app bars.
+
+_**NOTE: There is no explicit bottom drawer in Flutter but it can be created by calling `showModalBottomSheet` from a menu icon within a `BottomAppBar` in the `bottomNavigationBar` slot in the `Scaffold`.**_
+
+## Theming a navigation drawer
+
+Theming for a navigation drawer can be done by theming the widgets that are inside the `Drawer`.
+
+In the [Modal navigation drawer](#modal-navigation-drawer) above, the colors of the active destinations are based off the `ThemeData.primaryColor`.
```

Comment:
Since we are trying to move towards using the color scheme, maybe the example should be updated to use `ThemeData.colorScheme.primary` instead

Reply:
The tradeoff here would be to manually set the AppBar backgroundColor:

https://github.com/flutter/flutter/issues/67921

How about filing an issue here to update once that bug is fixed?

Reply:
Ah gotcha. I'm good with that.

An alternative, could be to hardcode the app bar's background color. But that isn't as clean